front/app/component-library/components/CollapsibleContainer/index.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { ReactNode, useState } from 'react';
+
+import { CSSTransition } from 'react-transition-group';
+import styled from 'styled-components';
+
+import useInstanceId from '../../hooks/useInstanceId';
+import { isRtl } from '../../utils/styleUtils';
+import Box, { BoxMarginProps, BoxPaddingProps, BoxWidthProps } from '../Box';
+import Icon from '../Icon';
+import Title, { Variant, FontSize, FontWeight } from '../Title';
+
+type CollapsibleContainerProps = {
+  children: ReactNode;
+  title: ReactNode;
+  isOpenByDefault?: boolean;
+  useRegionRole?: boolean;
+} & BoxMarginProps &
+  BoxWidthProps &
+  BoxPaddingProps &
+  TitleProps;
+
+type TitleProps = {
+  titleVariant: Variant;
+  titleFontSize?: FontSize;
+  titlePadding?: string;
+  titleFontWeight?: FontWeight;
+};
+
+const ChevronIcon = styled(Icon)`
+  transition: fill 80ms ease-out, transform 200ms ease-out;
+`;
+
+const TitleButton = styled.button`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  &.expanded {
+    ${ChevronIcon} {
+      transform: rotate(90deg);
+    }
+  }
+  &:hover {
+    ${ChevronIcon} {
+      fill: #000;
+    }
+  }
+
+  &:focus-visible {
+    outline: 2px solid black;
+    border-radius: 4px;
+  }
+
+  ${isRtl`
+    text-align: right;
+    direction: rtl;
+    ${ChevronIcon} {
+        transform: rotate(180deg);
+    }
+  `}
+`;
+
+const CollapseContainer = styled(Box)`
+  opacity: 1;
+  transition: 'all 1000ms cubic-bezier(0.165, 0.84, 0.44, 1)';
+  will-change: opacity, height;
+
+  ${isRtl`
+    text-align: right;
+    direction: rtl;
+`}
+
+  &.expanded-enter {
+    opacity: 0;
+    max-height: 0px;
+    overflow: hidden;
+
+    &.expanded-enter-active {
+      opacity: 1;
+      max-height: 1500px;
+      overflow: hidden;
+    }
+  }
+
+  &.expanded-enter-done {
+    opacity: 1;
+    overflow: visible;
+  }
+
+  &.expanded-exit {
+    opacity: 1;
+    max-height: 1500px;
+    overflow: hidden;
+
+    &.collapsed-exit-active {
+      opacity: 0;
+      max-height: 1500px;
+      overflow: hidden;
+    }
+  }
+`;
```

Comment:
Btw, I didn't check these styles. They could perhaps be trimmed as well but fine to leave for now. :)